pkgs/by-name/sp/spectral-language-server/package.nix

```
@@ -0,0 +1,93 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, typescript
+, jq
+}:
+let
+  # Instead of the build script that spectral-language-server provides (ref: https://github.com/luizcorreia/spectral-language-server/blob/master/script/vscode-spectral-build.sh), we build vscode-spectral manually.
+  vscodeSpectral = buildNpmPackage rec {
+    pname = "vscode-spectral";
+    version = "1.1.2";
+
+    src = fetchFromGitHub {
+      owner = "stoplightio";
+      repo = pname;
```

```suggestion
      repo = "vscode-spectral";
```
backrefrencing pname is discouraged, see https://github.com/NixOS/nixpkgs/issues/277994

---

It was really educational. Thank you!